File: Java/T4/codigos2021/T04_Dialogos/app/src/main/java/com/borja/t04_dialogos/dialogos/DialogoComunicar.java
package com.borja.t04_dialogos.dialogos;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class DialogoComunicar extends DialogFragment {

    private String titulo="Titulo";

    public static DialogoComunicar newInstance(String titulo) {

        Bundle bundle = new Bundle();
        DialogoComunicar dialogoComunicar = new DialogoComunicar();
        bundle.putString("titulo", titulo);
        dialogoComunicar.setArguments(bundle);
        return dialogoComunicar;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (this.getArguments() != null) {
            this.titulo = this.getArguments().getString("titulo");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(titulo);
        return builder.create();
    }
}

File: Kotlin/T5/codigos2223/T5_Dialogos/app/src/main/java/com/develop/t5_dialogos/dialogs/DialogoComunicar.kt
package com.develop.t5_dialogos.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class DialogoComunicar: DialogFragment() {

    private lateinit var nombre: String
    private lateinit var mensaje: String

    companion object{
        // todo lo declarado aqui es accesible directamente
        // 1. TODO creo un método newInstance como parametro lo que va a recibir el dialogo
        //      devuelve el objeto del tipo que quiere construir

        fun newInstance(nombre: String): DialogoComunicar{

            val dialogo = DialogoComunicar()
            // 2. TODO creo los argumentos con los datos que van en los parámetros
            // y se lo pongo al cuadro
            val bundle = Bundle()
            bundle.putString("nombre", nombre)
            bundle.putString("mensaje", "Esto es un ejemplo de mensaje pasado")
            dialogo.arguments = bundle
            return dialogo
        }
    }

    // cuadro de diálogo

    override fun onAttach(context: Context) {
        super.onAttach(context)
        // 3. TODO recupero todos los argumentos que quiera utilizar
        this.nombre =  this.arguments?.getString("nombre", "sobre el tag") ?: "Sin nombrw"
        this.mensaje =  this.arguments?.getString("mensaje", "sobre el tag") ?: "Sin nombrw"
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val builder =  AlertDialog.Builder(requireContext())
        builder.setTitle(nombre)
        builder.setMessage(mensaje)
        return builder.create()
    }

}